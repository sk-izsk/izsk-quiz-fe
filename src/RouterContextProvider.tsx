import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';

export interface RouterContextProviderProps {}

const RouterContextProvider: React.FC<RouterContextProviderProps> = ({ children }) => {
  const history: any = createMemoryHistory();
  return <Router history={history}>{children} </Router>;
};

export { RouterContextProvider };
