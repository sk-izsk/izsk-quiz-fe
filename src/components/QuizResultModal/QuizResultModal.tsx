import { makeStyles } from '@material-ui/core';
import { format } from 'date-fns';
import React from 'react';
import { TiDelete, TiTick } from 'react-icons/ti';
import { Link } from 'react-router-dom';
import { Button, CardAction, H5, H6, Subtitle2 } from 'ui-neumorphism';
import { DialogContainer } from '..';
import { CustomTheme, theme } from '../../theme/muiTheme';

export interface QuizResultModalProps {
  visible?: boolean;
  onClose?: () => void;
  retry?: () => void;
  title?: string;
  type?: string;
  date?: string | Date;
  totalQuestion?: number;
  correctAnswer?: number;
}

const useStyles = makeStyles((theme: CustomTheme) => ({
  modalContainer: {
    maxWidth: 500,
    width: '100%',
    padding: theme.spacing(1),
  },
  dateContainer: {
    marginTop: theme.spacing(1.5),
    marginBottom: theme.spacing(1.5),
    gridArea: 'date',
  },
  card: {
    padding: theme.spacing(1),
  },
  btnContainer: {
    display: 'grid',
    gridTemplateColumns: '1fr 1fr',
  },
  btn: {
    maxWidth: 200,
    width: '100%',
  },
  cardContent: {
    textAlign: 'center',
  },
  failedText: {
    color: theme.palette.secondary.main,
    fontWeight: 'bold',
  },
  passedText: {
    color: theme.palette.success.main,
    fontWeight: 'bold',
  },
  link: {
    textDecoration: 'none',
    '&:hover': {
      textDecoration: 'none',
    },
  },
  boldText: {
    fontWeight: 'bold',
  },
}));

const QuizResultModal: React.FC<QuizResultModalProps> = ({
  onClose,
  visible,
  correctAnswer,
  date,
  retry,
  title,
  totalQuestion,
  type,
}) => {
  const classes = useStyles();

  const quizResult = (correctAnswer as number) / (totalQuestion as number) >= 0.7 ? 'Passed' : 'Failed';

  return (
    <DialogContainer
      cardContentStyle={classes.cardContent}
      inset={true}
      visible={visible}
      onClose={onClose}
      cardAction={
        <CardAction className={classes.btnContainer}>
          <Link onClick={onClose} className={classes.link} to='/home'>
            <Button className={classes.btn} rounded color={theme.palette.primary.main}>
              Close
            </Button>
          </Link>
          {retry ? (
            <Button onClick={retry} className={classes.btn} rounded color={theme.palette.primary.main}>
              Retry
            </Button>
          ) : (
            <Button onClick={retry} className={classes.btn} rounded color={theme.palette.primary.main}>
              Re-take
            </Button>
          )}
        </CardAction>
      }
    >
      <H5 className={classes.boldText}>{title}</H5>
      <H6>
        Type of quiz: <span className={classes.boldText}>{type}</span>
      </H6>
      {quizResult === 'Passed' ? (
        <TiTick size={100} color={theme.palette.success.main} />
      ) : (
        <TiDelete size={100} color={theme.palette.secondary.main} />
      )}
      <Subtitle2 className={classes.dateContainer} secondary>
        {date && format(date as Date, 'do MMM yyyy')}
      </Subtitle2>
      <H6>
        Question answered correctly:
        <span className={classes.boldText}>
          {correctAnswer}/{totalQuestion}
        </span>
      </H6>
      <H6>
        Result of quiz :
        <span className={quizResult === 'Failed' ? classes.failedText : classes.passedText}> {quizResult}</span>
      </H6>
    </DialogContainer>
  );
};

export { QuizResultModal };
