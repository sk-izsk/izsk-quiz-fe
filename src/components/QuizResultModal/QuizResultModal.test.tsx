import { render } from '@testing-library/react';
import React from 'react';
import { RouterContextProvider } from '../../RouterContextProvider';
import { QuizResultModal } from './QuizResultModal';

const mockData = {
  visible: true,
  onClose: jest.fn(),
  retry: jest.fn(),
  title: 'mock title',
  type: 'mock type',
  totalQuestion: 20,
  correctAnswer: 10,
};

test('QuizResultModal renders correctly', () => {
  const { asFragment } = render(
    <RouterContextProvider>
      <QuizResultModal {...mockData} />
    </RouterContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
