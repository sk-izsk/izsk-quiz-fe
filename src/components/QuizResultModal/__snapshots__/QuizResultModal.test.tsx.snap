// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizResultModal renders correctly 1`] = `
<DocumentFragment>
   
</DocumentFragment>
`;
